docs/proposals/20210203-unmanaged-infrastructure.md

```diff
@@ -0,0 +1,170 @@
+[Tools for generating](https://github.com/ekalinin/github-markdown-toc) a table of contents from markdown are available.
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Requirements (Optional)](#requirements-optional)
+      - [Functional Requirements](#functional-requirements)
+        - [FR1](#fr1)
+        - [FR2](#fr2)
+      - [Non-Functional Requirements](#non-functional-requirements)
+        - [NFR1](#nfr1)
+        - [NFR2](#nfr2)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Security Model](#security-model)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan [optional]](#test-plan-optional)
+    - [Graduation Criteria [optional]](#graduation-criteria-optional)
+    - [Version Skew Strategy [optional]](#version-skew-strategy-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+### Managed cluster infrastructure
+
+Cluster infratructure which lifecycle is managed by a provider infraCluster CR.
+E.g in AWS:
+- Network
+    - VPC
+    - Subnets
+    - Internet gateways
+    - Nat gateways
+    - Route tables
+- Security groups
+- Load balancers
+
+### Unmanaged cluster infrastructure
+Cluster infratructure which lifecycle is not managed by CAPI but rather by an external entity.
+
+### Data plane
+Kubernetes endpoint accesible by a workload cluster admin which nodes capacity is intended exclusively to run user facing workloads
+
+## Summary
+
+This proposal introduces first class support for "unmanaged" infrastructure for CAPI providers to consolidate the boundaries between managed and unmanaged cluster infrastructure.
+
+## Motivation
+
+Currently CAPI providers support an opinionated happy path to create and manage cluster infrastructure lifecycle. The fundamental use case we want to support is bring your own infrastructure. An "unmanaged" CAPI infraCluster won't reconcile or manage the lifecycle of the cluster infrastructure, but CAPI will be able to create compute nodes within it.
+
+This will ease adoption of CAPI in heterogeneous real world environments with restricted privileges and where the provider infrastructure for the cluster needs to be managed out of band.
+
+### Goals
+
+- Introduce support for "unmanaged" infrastructure consistently across CAPI providers.
+- The machine controller must be able to operate and manage machines when the infastructure is "unmanaged".
```

Comment:
This seems an implementation detail, which I'm not totally in favor of. I'd rather try to bridge the behaviors and requirements rather than going down the path of requiring these controllers to change their behavior based on a filed, more concretely we should avoid having a bunch of `if <unmanaged> then do this; or do that`, instead we could consider a world where a contract is put in place where a controller _must_ be able to work in both modes consistently

Reply:
Let's discuss this here https://github.com/kubernetes-sigs/cluster-api/pull/4135#discussion_r569620731